### DISCUSSION TITLE
form action + event.locals question

### DISCUSSION THREAD

**Discussion by coryvirok** [sveltejs/kit#8222] [Q&A]
Is there a way to tell if a form action is invoked as part of a request from an enhanced form?

In my app I populate `event.locals` with a database object called "currentOrg". I have a form action that switches the current org. As per the docs, I'm updating `event.locals` in that form action, however depending on how the form action is being invoked this is not always necessary.

Scenario 1: basic (no enhanced form) POST

The order of ops seems to be:
1. hooks.server.js - `handle()` - sets `event.locals.currentOrg` based on cookie
2. form action invoked and sets `event.locals.currentOrg` to the new org + sets the cookie to contain the new org
3. +page.server.js - `load()`
4. +page.svelte rendered
5. -- done --

In this scenario, we definitely need the `event.locals.currentOrg` to be updated since the page's load function is being called afterward in order to render the page.

Scenario 2: enhanced form POST

1. hooks.server.js - `handle()` - sets `event.locals.currentOrg` based on cookie
2. form action invoked and response is returned as JSON to the browser (along with cookie containing new org)
3. -- done --
4. a NEW request is made to the server:
    1. hooks.server.js - `handle()`
    2. +page.server.js - `load()`
    3. +page.svelte rendered
    4. -- done --

In this scenario, there is no need to set `event.locals.currentOrg` in the form action since the request is completed and a brand new request is made to load the page.

Which begs the question - is there a way for a form action to know if it's being requested via an enhanced form? If there is, it would save some cycles to not have to look up the new org from the DB just to put it into `event.locals.currentOrg` right before the request is completed.

**Comment by TehBrian** [comment]
I'm running into the same-ish issue. Some data that goes to a form action is needed for an invalidated server page load. I could set it in cookies, but it'd be really nice if the form action's locals persisted to the next request.

I'm assuming remote functions will help solve this issue as the query function can be supplied with any necessary data as opposed to being restricted to data fed in by handle.
